File: src/services/instructions/loadstore/ST_S_MREGX_REGY.java
package emulator.source.loadstore

import emulator.engine.CpuContext
 import Instruction from '../Instruction'import REGISTER_VALUE_NAME_MAPPER from '@/constants/registers'

export default class ST_S_MREGX_REGY extends Instruction {
	public ST_S_MREGX_REGY(memory, address, source, 
 destination) {
		super(memory, address, source, destination)
		super.setAssembler("st.s [" + this.sdest + "], " + this.ssource)
	}

	
	exec ({ context, memory }) {
		context.memory[fix(context.getReg(this.destination) ) / 2] = (short) context.getReg(this.source) 
		
		context.pc  += 2
		updateViewer(context, fix(context.getReg(this.destination) ), context.getReg(this.source) )
	}
}

File: src/services/instructions/loadstore/StSMRegXRegY.js
import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination, ssource) => `st.s [${sdestination}], ${ssource}`

export default class StSMRegXRegY extends Instruction {
	constructor (memory, address, source, destination, symbolTable) {
		super(memory, address, source, destination, symbolTable)
		super.setAssembler(ASSEMBLER_INSTRUCTION_EXPRESSION(this.sdestination, this.ssource), symbolTable)
	}

	exec ({ context, memory }) {
		// TODO: -->> (short) context[REGISTER_VALUE_NAME_MAPPER[this.source]]
		const address = Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.destination]])
		const content = context[REGISTER_VALUE_NAME_MAPPER[this.source]]

		memory[Math.floor(address / 2)] = content
		context.pc  += 2

		return {
			address: address,
			content: content
		}
	}
}
